File: v2/pkg/runtime/window.go
package runtime

import (
	"context"

	"github.com/wailsapp/wails/v2/pkg/options"
)

// WindowSetTitle sets the title of the window
func WindowSetTitle(ctx context.Context, title string) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetTitle(title)
}

// WindowFullscreen makes the window fullscreen
func WindowFullscreen(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowFullscreen()
}

// WindowUnfullscreen makes the window UnFullscreen
func WindowUnfullscreen(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowUnfullscreen()
}

// WindowCenter the window on the current screen
func WindowCenter(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowCenter()
}

// WindowReload will reload the window contents
func WindowReload(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowReload()
}

// WindowReloadApp will reload the application
func WindowReloadApp(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowReloadApp()
}

func WindowSetSystemDefaultTheme(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetSystemDefaultTheme()
}

func WindowSetLightTheme(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetLightTheme()
}

func WindowSetDarkTheme(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetDarkTheme()
}

// WindowShow shows the window if hidden
func WindowShow(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowShow()
}

// WindowHide the window
func WindowHide(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowHide()
}

// WindowSetSize sets the size of the window
func WindowSetSize(ctx context.Context, width int, height int) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetSize(width, height)
}

func WindowGetSize(ctx context.Context) (int, int) {
	appFrontend := getFrontend(ctx)
	return appFrontend.WindowGetSize()
}

// WindowSetMinSize sets the minimum size of the window
func WindowSetMinSize(ctx context.Context, width int, height int) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetMinSize(width, height)
}

// WindowSetMaxSize sets the maximum size of the window
func WindowSetMaxSize(ctx context.Context, width int, height int) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetMaxSize(width, height)
}

// WindowSetAlwaysOnTop sets the window AlwaysOnTop or not on top
func WindowSetAlwaysOnTop(ctx context.Context, b bool) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetAlwaysOnTop(b)
}

// WindowSetPosition sets the position of the window
func WindowSetPosition(ctx context.Context, x int, y int) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowSetPosition(x, y)
}

func WindowGetPosition(ctx context.Context) (int, int) {
	appFrontend := getFrontend(ctx)
	return appFrontend.WindowGetPosition()
}

// WindowMaximise the window
func WindowMaximise(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowMaximise()
}

// WindowToggleMaximise the window
func WindowToggleMaximise(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowToggleMaximise()
}

// WindowUnmaximise the window
func WindowUnmaximise(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowUnmaximise()
}

// WindowMinimise the window
func WindowMinimise(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowMinimise()
}

// WindowUnminimise the window
func WindowUnminimise(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowUnminimise()
}

// WindowIsFullscreen get the window state is window Fullscreen
func WindowIsFullscreen(ctx context.Context) bool {
	appFrontend := getFrontend(ctx)
	return appFrontend.WindowIsFullscreen()
}

// WindowIsMaximised get the window state is window Maximised
func WindowIsMaximised(ctx context.Context) bool {
	appFrontend := getFrontend(ctx)
	return appFrontend.WindowIsMaximised()
}

// WindowIsMinimised get the window state is window Minimised
func WindowIsMinimised(ctx context.Context) bool {
	appFrontend := getFrontend(ctx)
	return appFrontend.WindowIsMinimised()
}

// WindowIsNormal get the window state is window Normal
func WindowIsNormal(ctx context.Context) bool {
	appFrontend := getFrontend(ctx)
	return appFrontend.WindowIsNormal()
}

// WindowExecJS executes the given Js in the window
func WindowExecJS(ctx context.Context, js string) {
	appFrontend := getFrontend(ctx)
	appFrontend.ExecJS(js)
}

func WindowSetBackgroundColour(ctx context.Context, R, G, B, A uint8) {
	appFrontend := getFrontend(ctx)
	col := &options.RGBA{
		R: R,
		G: G,
		B: B,
		A: A,
	}
	appFrontend.WindowSetBackgroundColour(col)
}

func WindowPrint(ctx context.Context) {
	appFrontend := getFrontend(ctx)
	appFrontend.WindowPrint()
}

File: v2/internal/frontend/runtime/desktop/window.js
/*
 _	   __	  _ __
| |	 / /___ _(_) /____
| | /| / / __ `/ / / ___/
| |/ |/ / /_/ / / (__  )
|__/|__/\__,_/_/_/____/
The electron alternative for Go
(c) Lea Anthony 2019-present
*/

/* jshint esversion: 9 */


import {Call} from "./calls";

export function WindowReload() {
    window.location.reload();
}

export function WindowReloadApp() {
    window.WailsInvoke('WR');
}

export function WindowSetSystemDefaultTheme() {
    window.WailsInvoke('WASDT');
}

export function WindowSetLightTheme() {
    window.WailsInvoke('WALT');
}

export function WindowSetDarkTheme() {
    window.WailsInvoke('WADT');
}

/**
 * Place the window in the center of the screen
 *
 * @export
 */
export function WindowCenter() {
    window.WailsInvoke('Wc');
}

/**
 * Sets the window title
 *
 * @param {string} title
 * @export
 */
export function WindowSetTitle(title) {
    window.WailsInvoke('WT' + title);
}

/**
 * Makes the window go fullscreen
 *
 * @export
 */
export function WindowFullscreen() {
    window.WailsInvoke('WF');
}

/**
 * Reverts the window from fullscreen
 *
 * @export
 */
export function WindowUnfullscreen() {
    window.WailsInvoke('Wf');
}

/**
 * Returns the state of the window, i.e. whether the window is in full screen mode or not.
 *
 * @export
 * @return {Promise<boolean>} The state of the window
 */
export function WindowIsFullscreen() {
    return Call(":wails:WindowIsFullscreen");
}

/**
 * Set the Size of the window
 *
 * @export
 * @param {number} width
 * @param {number} height
 */
export function WindowSetSize(width, height) {
    window.WailsInvoke('Ws:' + width + ':' + height);
}

/**
 * Get the Size of the window
 *
 * @export
 * @return {Promise<{w: number, h: number}>} The size of the window

 */
export function WindowGetSize() {
    return Call(":wails:WindowGetSize");
}

/**
 * Set the maximum size of the window
 *
 * @export
 * @param {number} width
 * @param {number} height
 */
export function WindowSetMaxSize(width, height) {
    window.WailsInvoke('WZ:' + width + ':' + height);
}

/**
 * Set the minimum size of the window
 *
 * @export
 * @param {number} width
 * @param {number} height
 */
export function WindowSetMinSize(width, height) {
    window.WailsInvoke('Wz:' + width + ':' + height);
}



/**
 * Set the window AlwaysOnTop or not on top
 *
 * @export
 */
export function WindowSetAlwaysOnTop(b) {

    window.WailsInvoke('WATP:' + (b ? '1' : '0'));
}




/**
 * Set the Position of the window
 *
 * @export
 * @param {number} x
 * @param {number} y
 */
export function WindowSetPosition(x, y) {
    window.WailsInvoke('Wp:' + x + ':' + y);
}

/**
 * Get the Position of the window
 *
 * @export
 * @return {Promise<{x: number, y: number}>} The position of the window
 */
export function WindowGetPosition() {
    return Call(":wails:WindowGetPos");
}

/**
 * Hide the Window
 *
 * @export
 */
export function WindowHide() {
    window.WailsInvoke('WH');
}

/**
 * Show the Window
 *
 * @export
 */
export function WindowShow() {
    window.WailsInvoke('WS');
}

/**
 * Maximise the Window
 *
 * @export
 */
export function WindowMaximise() {
    window.WailsInvoke('WM');
}

/**
 * Toggle the Maximise of the Window
 *
 * @export
 */
export function WindowToggleMaximise() {
    window.WailsInvoke('Wt');
}

/**
 * Unmaximise the Window
 *
 * @export
 */
export function WindowUnmaximise() {
    window.WailsInvoke('WU');
}

/**
 * Returns the state of the window, i.e. whether the window is maximised or not.
 *
 * @export
 * @return {Promise<boolean>} The state of the window
 */
export function WindowIsMaximised() {
    return Call(":wails:WindowIsMaximised");
}

/**
 * Minimise the Window
 *
 * @export
 */
export function WindowMinimise() {
    window.WailsInvoke('Wm');
}

/**
 * Unminimise the Window
 *
 * @export
 */
export function WindowUnminimise() {
    window.WailsInvoke('Wu');
}

/**
 * Returns the state of the window, i.e. whether the window is minimised or not.
 *
 * @export
 * @return {Promise<boolean>} The state of the window
 */
export function WindowIsMinimised() {
    return Call(":wails:WindowIsMinimised");
}

/**
 * Returns the state of the window, i.e. whether the window is normal or not.
 *
 * @export
 * @return {Promise<boolean>} The state of the window
 */
export function WindowIsNormal() {
    return Call(":wails:WindowIsNormal");
}

/**
 * Sets the background colour of the window
 *
 * @export
 * @param {number} R Red
 * @param {number} G Green
 * @param {number} B Blue
 * @param {number} A Alpha
 */
export function WindowSetBackgroundColour(R, G, B, A) {
    let rgba = JSON.stringify({r: R || 0, g: G || 0, b: B || 0, a: A || 255});
    window.WailsInvoke('Wr:' + rgba);
}

